archive/finder.go

```diff
@@ -11,83 +11,112 @@ import (
 	"github.com/brimsec/zq/zng/resolver"
 )
 
+type findOptions struct {
+	skipMissing bool
+	zctx        *resolver.Context
+	addPath     func(ark *Archive, si SpanInfo, rec *zng.Record) (*zng.Record, error)
+}
+
+type FindOption func(*findOptions) error
```

Comment:
You are exporting a function with an un-exported  argument.

Reply:
I think you are perhaps looking to do something similar to what zap does with options? https://github.com/uber-go/zap/blob/master/options.go#L30

Reply:
That's intended: it allows users outside of the package write code like:
```go
	var findOptions []archive.FindOption
	if c.pathField != "" {
		findOptions = append(findOptions, archive.AddPath(c.pathField, true))
	}
```
to create slices of `FindOption`'s. However, they don't need access to the internal workings of the `findOptions` type.

Reply:
(I pointed @mattnibs to dave cheney's blog post on functional options in go: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis )